scripts_python/ae_train_model.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Train a model for Autoencoders
+"""
+import argparse
+import logging
+import os
+
+# comet_ml not used, but comet_ml requires to be imported before torch.
+# See bug report here https://github.com/Lightning-AI/lightning/issues/5829
+# Importing now to solve issues later.
+import comet_ml # noqa F401
+import torch
+
+from scilpy.io.utils import (assert_inputs_exist, assert_outputs_exist,
+                             add_verbose_arg)
+
+from dwi_ml.data.dataset.utils import prepare_multisubjectdataset
+from dwi_ml.experiment_utils.prints import format_dict_to_str
+from dwi_ml.experiment_utils.timer import Timer
+from dwi_ml.io_utils import add_memory_args
+from dwi_ml.models.projects.ae_models import ModelAE
+from dwi_ml.training.trainers import DWIMLAbstractTrainer
+from dwi_ml.training.utils.batch_samplers import (add_args_batch_sampler,
+                                                  prepare_batch_sampler)
+from dwi_ml.training.utils.batch_loaders import (add_args_batch_loader)
+from dwi_ml.training.utils.trainer import (add_training_args, run_experiment,
+                                           format_lr)
+from dwi_ml.training.batch_loaders import DWIMLStreamlinesBatchLoader
+from dwi_ml.training.utils.experiment import (
+    add_mandatory_args_experiment_and_hdf5_path)
+
+
+def prepare_arg_parser():
+    p = argparse.ArgumentParser(description=__doc__,
+                                formatter_class=argparse.RawTextHelpFormatter)
+    add_mandatory_args_experiment_and_hdf5_path(p)
+    add_args_batch_sampler(p)
+    add_args_batch_loader(p)
+    add_training_args(p)
```

Comment:
Why not put the kernel_size, latent_space_dims as parameters if they are parameters to the `__init__` of your model?

Else, I think they should not be parameters, they should be hard-coded like all the other dimensions of your CNN steps.